lib/recognize-stream.ts

```diff
@@ -241,12 +242,18 @@ class RecognizeStream extends Duplex {
     // for the last argument, `tlsOptions` gets passed to Node's `http` library,
     // which allows us to pass a rejectUnauthorized option
     // for disabling SSL verification (for ICP)
+
+    // add custom agent in the request options if given by user
+    // default request options to null
+    const { agent } = options;
+    const requestOptions = agent ? { agent } : null;
+
     const socket = (this.socket = new w3cWebSocket(
       url,
       null,
       null,
       options.headers,
-      null,
+      requestOptions,
```

Comment:
I guess the change here is based on https://github.com/theturtle32/WebSocket-Node/blob/master/docs/WebSocketClient.md#connect-using-a-proxy-server

Reply:
Yep, that's correct!